// Control.Comonad.Store.Class/index.js
"use strict";
var Control_Comonad = require("Control.Comonad");
var Prelude = require("Prelude");
var Control_Extend = require("Control.Extend");
var Control_Comonad_Store_Trans = require("Control.Comonad.Store.Trans");
var Data_Tuple = require("Data.Tuple");
function ComonadStore(__superclass_Control$dotComonad$dotComonad_0, peek, pos) {
    this["__superclass_Control.Comonad.Comonad_0"] = __superclass_Control$dotComonad$dotComonad_0;
    this.peek = peek;
    this.pos = pos;
};
var pos = function (dict) {
    return dict.pos;
};
var peek = function (dict) {
    return dict.peek;
};
var peeks = function (__dict_ComonadStore_0) {
    return function (f) {
        return function (x) {
            return peek(__dict_ComonadStore_0)(f(pos(__dict_ComonadStore_0)(x)))(x);
        };
    };
};
var seeks = function (__dict_ComonadStore_1) {
    return function (__dict_Extend_2) {
        return function (f) {
            return function (x) {
                return peeks(__dict_ComonadStore_1)(f)(Control_Extend.duplicate(__dict_Extend_2)(x));
            };
        };
    };
};
var seek = function (__dict_ComonadStore_3) {
    return function (__dict_Extend_4) {
        return function (s) {
            return function (x) {
                return peek(__dict_ComonadStore_3)(s)(Control_Extend.duplicate(__dict_Extend_4)(x));
            };
        };
    };
};
var experiment = function (__dict_ComonadStore_5) {
    return function (__dict_Functor_6) {
        return function (f) {
            return function (x) {
                return Prelude["<$>"](__dict_Functor_6)(Prelude.flip(peek(__dict_ComonadStore_5))(x))(f(pos(__dict_ComonadStore_5)(x)));
            };
        };
    };
};
var comonadStoreStoreT = function (__dict_Comonad_7) {
    return new ComonadStore(function () {
        return Control_Comonad_Store_Trans.comonadStoreT(__dict_Comonad_7);
    }, function (_207) {
        return function (_208) {
            return Control_Comonad.extract(__dict_Comonad_7)(_208.value0)(_207);
        };
    }, function (_206) {
        return _206.value1;
    });
};
module.exports = {
    ComonadStore: ComonadStore, 
    comonadStoreStoreT: comonadStoreStoreT, 
    experiment: experiment, 
    peek: peek, 
    peeks: peeks, 
    pos: pos, 
    seek: seek, 
    seeks: seeks
};
