// Thermite.Html.Attributes/index.js
"use strict";
var Prelude = require("Prelude");

  function attribute(attr) {
    return function(value) {
      return [attr, value];
    };
  }   
  ;
var wmode = attribute("wmode");
var width = attribute("width");
var value = attribute("value");
var useMap = attribute("useMap");
var title = attribute("title");
var target = attribute("target");
var tabIndex = attribute("tabIndex");
var step = attribute("step");
var start = attribute("start");
var srcSet = attribute("srcSet");
var srcDoc = attribute("srcDoc");
var src = attribute("src");
var spellCheck = attribute("spellCheck");
var span = attribute("span");
var sizes = attribute("sizes");
var size = attribute("size");
var shape = attribute("shape");
var selected = attribute("selected");
var seamless = attribute("seamless");
var scrolling = attribute("scrolling");
var scope = attribute("scope");
var sandbox = attribute("sandbox");
var rows = attribute("rows");
var rowSpan = attribute("rowSpan");
var role = attribute("role");
var required = attribute("required");
var rel = attribute("rel");
var readOnly = attribute("readOnly");
var radioGroup = attribute("radioGroup");
var preload = attribute("preload");
var poster = attribute("poster");
var placeholder = attribute("placeholder");
var pattern = attribute("pattern");
var open = attribute("open");
var noValidate = attribute("noValidate");
var name = attribute("name");
var muted = attribute("muted");
var multiple = attribute("multiple");
var min = attribute("min");
var method = attribute("method");
var mediaGroup = attribute("mediaGroup");
var media = attribute("media");
var maxLength = attribute("maxLength");
var max = attribute("max");
var marginWidth = attribute("marginWidth");
var marginHeight = attribute("marginHeight");
var manifest = attribute("manifest");
var loop = attribute("loop");
var list = attribute("list");
var lang = attribute("lang");
var label = attribute("label");
var icon = attribute("icon");
var httpEquiv = attribute("httpEquiv");
var htmlFor = attribute("htmlFor");
var hrefLang = attribute("hrefLang");
var href = attribute("href");
var hidden = attribute("hidden");
var height = attribute("height");
var frameBorder = attribute("frameBorder");
var formTarget = attribute("formTarget");
var formNoValidate = attribute("formNoValidate");
var formMethod = attribute("formMethod");
var formEncType = attribute("formEncType");
var formAction = attribute("formAction");
var form = attribute("form");
var encType = attribute("encType");
var draggable = attribute("draggable");
var download = attribute("download");
var disabled = attribute("disabled");
var dir = attribute("dir");
var defer = attribute("defer");
var dateTime = attribute("dateTime");
var crossOrigin = attribute("crossOrigin");
var coords = attribute("coords");
var controls = attribute("controls");
var contextMenu = attribute("contextMenu");
var contentEditable = attribute("contentEditable");
var content = attribute("content");
var cols = attribute("cols");
var colSpan = attribute("colSpan");
var className = attribute("className");
var classID = attribute("classID");
var checked = attribute("checked");
var charSet = attribute("charSet");
var cellSpacing = attribute("cellSpacing");
var cellPadding = attribute("cellPadding");
var autoPlay = attribute("autoPlay");
var autoComplete = attribute("autoComplete");
var async = attribute("async");
var alt = attribute("alt");
var allowTransparency = attribute("allowTransparency");
var allowFullScreen = attribute("allowFullScreen");
var action = attribute("action");
var accessKey = attribute("accessKey");
var acceptCharset = attribute("acceptCharset");
var accept = attribute("accept");
var _type = attribute("type");
var _id = attribute("id");
module.exports = {
    _id: _id, 
    _type: _type, 
    accept: accept, 
    acceptCharset: acceptCharset, 
    accessKey: accessKey, 
    action: action, 
    allowFullScreen: allowFullScreen, 
    allowTransparency: allowTransparency, 
    alt: alt, 
    async: async, 
    attribute: attribute, 
    autoComplete: autoComplete, 
    autoPlay: autoPlay, 
    cellPadding: cellPadding, 
    cellSpacing: cellSpacing, 
    charSet: charSet, 
    checked: checked, 
    classID: classID, 
    className: className, 
    colSpan: colSpan, 
    cols: cols, 
    content: content, 
    contentEditable: contentEditable, 
    contextMenu: contextMenu, 
    controls: controls, 
    coords: coords, 
    crossOrigin: crossOrigin, 
    dateTime: dateTime, 
    defer: defer, 
    dir: dir, 
    disabled: disabled, 
    download: download, 
    draggable: draggable, 
    encType: encType, 
    form: form, 
    formAction: formAction, 
    formEncType: formEncType, 
    formMethod: formMethod, 
    formNoValidate: formNoValidate, 
    formTarget: formTarget, 
    frameBorder: frameBorder, 
    height: height, 
    hidden: hidden, 
    href: href, 
    hrefLang: hrefLang, 
    htmlFor: htmlFor, 
    httpEquiv: httpEquiv, 
    icon: icon, 
    label: label, 
    lang: lang, 
    list: list, 
    loop: loop, 
    manifest: manifest, 
    marginHeight: marginHeight, 
    marginWidth: marginWidth, 
    max: max, 
    maxLength: maxLength, 
    media: media, 
    mediaGroup: mediaGroup, 
    method: method, 
    min: min, 
    multiple: multiple, 
    muted: muted, 
    name: name, 
    noValidate: noValidate, 
    open: open, 
    pattern: pattern, 
    placeholder: placeholder, 
    poster: poster, 
    preload: preload, 
    radioGroup: radioGroup, 
    readOnly: readOnly, 
    rel: rel, 
    required: required, 
    role: role, 
    rowSpan: rowSpan, 
    rows: rows, 
    sandbox: sandbox, 
    scope: scope, 
    scrolling: scrolling, 
    seamless: seamless, 
    selected: selected, 
    shape: shape, 
    size: size, 
    sizes: sizes, 
    span: span, 
    spellCheck: spellCheck, 
    src: src, 
    srcDoc: srcDoc, 
    srcSet: srcSet, 
    start: start, 
    step: step, 
    tabIndex: tabIndex, 
    target: target, 
    title: title, 
    useMap: useMap, 
    value: value, 
    width: width, 
    wmode: wmode
};
