// Thermite.Html.Attributes/index.js
"use strict";
var Thermite_Internal = require("Thermite.Internal");
var Prelude = require("Prelude");
var wmode = Thermite_Internal.unsafeAttribute("wmode");
var width = Thermite_Internal.unsafeAttribute("width");
var value = Thermite_Internal.unsafeAttribute("value");
var useMap = Thermite_Internal.unsafeAttribute("useMap");
var title = Thermite_Internal.unsafeAttribute("title");
var target = Thermite_Internal.unsafeAttribute("target");
var tabIndex = Thermite_Internal.unsafeAttribute("tabIndex");
var step = Thermite_Internal.unsafeAttribute("step");
var start = Thermite_Internal.unsafeAttribute("start");
var srcSet = Thermite_Internal.unsafeAttribute("srcSet");
var srcDoc = Thermite_Internal.unsafeAttribute("srcDoc");
var src = Thermite_Internal.unsafeAttribute("src");
var spellCheck = Thermite_Internal.unsafeAttribute("spellCheck");
var span = Thermite_Internal.unsafeAttribute("span");
var sizes = Thermite_Internal.unsafeAttribute("sizes");
var size = Thermite_Internal.unsafeAttribute("size");
var shape = Thermite_Internal.unsafeAttribute("shape");
var selected = Thermite_Internal.unsafeAttribute("selected");
var seamless = Thermite_Internal.unsafeAttribute("seamless");
var scrolling = Thermite_Internal.unsafeAttribute("scrolling");
var scope = Thermite_Internal.unsafeAttribute("scope");
var sandbox = Thermite_Internal.unsafeAttribute("sandbox");
var rows = Thermite_Internal.unsafeAttribute("rows");
var rowSpan = Thermite_Internal.unsafeAttribute("rowSpan");
var role = Thermite_Internal.unsafeAttribute("role");
var required = Thermite_Internal.unsafeAttribute("required");
var rel = Thermite_Internal.unsafeAttribute("rel");
var readOnly = Thermite_Internal.unsafeAttribute("readOnly");
var radioGroup = Thermite_Internal.unsafeAttribute("radioGroup");
var preload = Thermite_Internal.unsafeAttribute("preload");
var poster = Thermite_Internal.unsafeAttribute("poster");
var placeholder = Thermite_Internal.unsafeAttribute("placeholder");
var pattern = Thermite_Internal.unsafeAttribute("pattern");
var open = Thermite_Internal.unsafeAttribute("open");
var noValidate = Thermite_Internal.unsafeAttribute("noValidate");
var name = Thermite_Internal.unsafeAttribute("name");
var muted = Thermite_Internal.unsafeAttribute("muted");
var multiple = Thermite_Internal.unsafeAttribute("multiple");
var min = Thermite_Internal.unsafeAttribute("min");
var method = Thermite_Internal.unsafeAttribute("method");
var mediaGroup = Thermite_Internal.unsafeAttribute("mediaGroup");
var media = Thermite_Internal.unsafeAttribute("media");
var maxLength = Thermite_Internal.unsafeAttribute("maxLength");
var max = Thermite_Internal.unsafeAttribute("max");
var marginWidth = Thermite_Internal.unsafeAttribute("marginWidth");
var marginHeight = Thermite_Internal.unsafeAttribute("marginHeight");
var manifest = Thermite_Internal.unsafeAttribute("manifest");
var loop = Thermite_Internal.unsafeAttribute("loop");
var list = Thermite_Internal.unsafeAttribute("list");
var lang = Thermite_Internal.unsafeAttribute("lang");
var label = Thermite_Internal.unsafeAttribute("label");
var icon = Thermite_Internal.unsafeAttribute("icon");
var httpEquiv = Thermite_Internal.unsafeAttribute("httpEquiv");
var htmlFor = Thermite_Internal.unsafeAttribute("htmlFor");
var hrefLang = Thermite_Internal.unsafeAttribute("hrefLang");
var href = Thermite_Internal.unsafeAttribute("href");
var hidden = Thermite_Internal.unsafeAttribute("hidden");
var height = Thermite_Internal.unsafeAttribute("height");
var frameBorder = Thermite_Internal.unsafeAttribute("frameBorder");
var formTarget = Thermite_Internal.unsafeAttribute("formTarget");
var formNoValidate = Thermite_Internal.unsafeAttribute("formNoValidate");
var formMethod = Thermite_Internal.unsafeAttribute("formMethod");
var formEncType = Thermite_Internal.unsafeAttribute("formEncType");
var formAction = Thermite_Internal.unsafeAttribute("formAction");
var form = Thermite_Internal.unsafeAttribute("form");
var encType = Thermite_Internal.unsafeAttribute("encType");
var draggable = Thermite_Internal.unsafeAttribute("draggable");
var download = Thermite_Internal.unsafeAttribute("download");
var disabled = Thermite_Internal.unsafeAttribute("disabled");
var dir = Thermite_Internal.unsafeAttribute("dir");
var defer = Thermite_Internal.unsafeAttribute("defer");
var dateTime = Thermite_Internal.unsafeAttribute("dateTime");
var crossOrigin = Thermite_Internal.unsafeAttribute("crossOrigin");
var coords = Thermite_Internal.unsafeAttribute("coords");
var controls = Thermite_Internal.unsafeAttribute("controls");
var contextMenu = Thermite_Internal.unsafeAttribute("contextMenu");
var contentEditable = Thermite_Internal.unsafeAttribute("contentEditable");
var content = Thermite_Internal.unsafeAttribute("content");
var cols = Thermite_Internal.unsafeAttribute("cols");
var colSpan = Thermite_Internal.unsafeAttribute("colSpan");
var className = Thermite_Internal.unsafeAttribute("className");
var classID = Thermite_Internal.unsafeAttribute("classID");
var checked = Thermite_Internal.unsafeAttribute("checked");
var charSet = Thermite_Internal.unsafeAttribute("charSet");
var cellSpacing = Thermite_Internal.unsafeAttribute("cellSpacing");
var cellPadding = Thermite_Internal.unsafeAttribute("cellPadding");
var autoPlay = Thermite_Internal.unsafeAttribute("autoPlay");
var autoFocus = Thermite_Internal.unsafeAttribute("autoFocus");
var autoComplete = Thermite_Internal.unsafeAttribute("autoComplete");
var async = Thermite_Internal.unsafeAttribute("async");
var alt = Thermite_Internal.unsafeAttribute("alt");
var allowTransparency = Thermite_Internal.unsafeAttribute("allowTransparency");
var allowFullScreen = Thermite_Internal.unsafeAttribute("allowFullScreen");
var action = Thermite_Internal.unsafeAttribute("action");
var accessKey = Thermite_Internal.unsafeAttribute("accessKey");
var acceptCharset = Thermite_Internal.unsafeAttribute("acceptCharset");
var accept = Thermite_Internal.unsafeAttribute("accept");
var _type = Thermite_Internal.unsafeAttribute("type");
var _id = Thermite_Internal.unsafeAttribute("id");
module.exports = {
    _id: _id, 
    _type: _type, 
    accept: accept, 
    acceptCharset: acceptCharset, 
    accessKey: accessKey, 
    action: action, 
    allowFullScreen: allowFullScreen, 
    allowTransparency: allowTransparency, 
    alt: alt, 
    async: async, 
    autoComplete: autoComplete, 
    autoFocus: autoFocus, 
    autoPlay: autoPlay, 
    cellPadding: cellPadding, 
    cellSpacing: cellSpacing, 
    charSet: charSet, 
    checked: checked, 
    classID: classID, 
    className: className, 
    colSpan: colSpan, 
    cols: cols, 
    content: content, 
    contentEditable: contentEditable, 
    contextMenu: contextMenu, 
    controls: controls, 
    coords: coords, 
    crossOrigin: crossOrigin, 
    dateTime: dateTime, 
    defer: defer, 
    dir: dir, 
    disabled: disabled, 
    download: download, 
    draggable: draggable, 
    encType: encType, 
    form: form, 
    formAction: formAction, 
    formEncType: formEncType, 
    formMethod: formMethod, 
    formNoValidate: formNoValidate, 
    formTarget: formTarget, 
    frameBorder: frameBorder, 
    height: height, 
    hidden: hidden, 
    href: href, 
    hrefLang: hrefLang, 
    htmlFor: htmlFor, 
    httpEquiv: httpEquiv, 
    icon: icon, 
    label: label, 
    lang: lang, 
    list: list, 
    loop: loop, 
    manifest: manifest, 
    marginHeight: marginHeight, 
    marginWidth: marginWidth, 
    max: max, 
    maxLength: maxLength, 
    media: media, 
    mediaGroup: mediaGroup, 
    method: method, 
    min: min, 
    multiple: multiple, 
    muted: muted, 
    name: name, 
    noValidate: noValidate, 
    open: open, 
    pattern: pattern, 
    placeholder: placeholder, 
    poster: poster, 
    preload: preload, 
    radioGroup: radioGroup, 
    readOnly: readOnly, 
    rel: rel, 
    required: required, 
    role: role, 
    rowSpan: rowSpan, 
    rows: rows, 
    sandbox: sandbox, 
    scope: scope, 
    scrolling: scrolling, 
    seamless: seamless, 
    selected: selected, 
    shape: shape, 
    size: size, 
    sizes: sizes, 
    span: span, 
    spellCheck: spellCheck, 
    src: src, 
    srcDoc: srcDoc, 
    srcSet: srcSet, 
    start: start, 
    step: step, 
    tabIndex: tabIndex, 
    target: target, 
    title: title, 
    useMap: useMap, 
    value: value, 
    width: width, 
    wmode: wmode
};
