// Thermite.Html.Elements/index.js
"use strict";
var Thermite_Html = require("Thermite.Html");
var Prelude = require("Prelude");
var wbr = Thermite_Html.createElement("body");
var wbr$prime = wbr([  ]);
var video = Thermite_Html.createElement("video");
var video$prime = video([  ]);
var $$var = Thermite_Html.createElement("var");
var var$prime = $$var([  ]);
var ul = Thermite_Html.createElement("ul");
var ul$prime = ul([  ]);
var u = Thermite_Html.createElement("u");
var u$prime = u([  ]);
var track = Thermite_Html.createElement("track");
var track$prime = track([  ]);
var tr = Thermite_Html.createElement("tr");
var tr$prime = tr([  ]);
var title = Thermite_Html.createElement("title");
var title$prime = title([  ]);
var time = Thermite_Html.createElement("time");
var time$prime = time([  ]);
var thead = Thermite_Html.createElement("thead");
var thead$prime = thead([  ]);
var th = Thermite_Html.createElement("th");
var th$prime = th([  ]);
var tfoot = Thermite_Html.createElement("tfoot");
var tfoot$prime = tfoot([  ]);
var textarea = Thermite_Html.createElement("textarea");
var textarea$prime = textarea([  ]);
var td = Thermite_Html.createElement("td");
var td$prime = td([  ]);
var tbody = Thermite_Html.createElement("tbody");
var tbody$prime = tbody([  ]);
var table = Thermite_Html.createElement("table");
var table$prime = table([  ]);
var sup = Thermite_Html.createElement("sup");
var sup$prime = sup([  ]);
var summary = Thermite_Html.createElement("summary");
var summary$prime = summary([  ]);
var sub = Thermite_Html.createElement("sub");
var sub$prime = sub([  ]);
var style = Thermite_Html.createElement("style");
var style$prime = style([  ]);
var strong = Thermite_Html.createElement("strong");
var strong$prime = strong([  ]);
var span = Thermite_Html.createElement("span");
var span$prime = span([  ]);
var source = Thermite_Html.createElement("source");
var source$prime = source([  ]);
var small = Thermite_Html.createElement("small");
var small$prime = small([  ]);
var select = Thermite_Html.createElement("select");
var select$prime = select([  ]);
var section = Thermite_Html.createElement("section");
var section$prime = section([  ]);
var script = Thermite_Html.createElement("script");
var script$prime = script([  ]);
var samp = Thermite_Html.createElement("samp");
var samp$prime = samp([  ]);
var s = Thermite_Html.createElement("s");
var s$prime = s([  ]);
var ruby = Thermite_Html.createElement("ruby");
var ruby$prime = ruby([  ]);
var rt = Thermite_Html.createElement("rt");
var rt$prime = rt([  ]);
var rp = Thermite_Html.createElement("rp");
var rp$prime = rp([  ]);
var q = Thermite_Html.createElement("q");
var q$prime = q([  ]);
var progress = Thermite_Html.createElement("progress");
var progress$prime = progress([  ]);
var pre = Thermite_Html.createElement("pre");
var pre$prime = pre([  ]);
var picture = Thermite_Html.createElement("picture");
var picture$prime = picture([  ]);
var param = Thermite_Html.createElement("param");
var param$prime = param([  ]);
var p = Thermite_Html.createElement("p");
var p$prime = p([  ]);
var output = Thermite_Html.createElement("output");
var output$prime = output([  ]);
var option = Thermite_Html.createElement("option");
var option$prime = option([  ]);
var optgroup = Thermite_Html.createElement("optgroup");
var optgroup$prime = optgroup([  ]);
var ol = Thermite_Html.createElement("ol");
var ol$prime = ol([  ]);
var object = Thermite_Html.createElement("object");
var object$prime = object([  ]);
var noscript = Thermite_Html.createElement("noscript");
var noscript$prime = noscript([  ]);
var nav = Thermite_Html.createElement("nav");
var nav$prime = nav([  ]);
var meter = Thermite_Html.createElement("meter");
var meter$prime = meter([  ]);
var meta = Thermite_Html.createElement("meta");
var meta$prime = meta([  ]);
var menuitem = Thermite_Html.createElement("menuitem");
var menuitem$prime = menuitem([  ]);
var menu = Thermite_Html.createElement("menu");
var menu$prime = menu([  ]);
var mark = Thermite_Html.createElement("mark");
var mark$prime = mark([  ]);
var map = Thermite_Html.createElement("map");
var map$prime = map([  ]);
var main = Thermite_Html.createElement("main");
var main$prime = main([  ]);
var link = Thermite_Html.createElement("link");
var li = Thermite_Html.createElement("li");
var li$prime = li([  ]);
var legend = Thermite_Html.createElement("legend");
var legend$prime = legend([  ]);
var label = Thermite_Html.createElement("label");
var label$prime = label([  ]);
var keygen = Thermite_Html.createElement("keygen");
var keygen$prime = keygen([  ]);
var kbd = Thermite_Html.createElement("kbd");
var kbd$prime = kbd([  ]);
var ins = Thermite_Html.createElement("ins");
var ins$prime = ins([  ]);
var input = Thermite_Html.createElement("input");
var input$prime = input([  ]);
var img = Thermite_Html.createElement("img");
var img$prime = img([  ]);
var iframe = Thermite_Html.createElement("iframe");
var iframe$prime = iframe([  ]);
var i = Thermite_Html.createElement("i");
var i$prime = i([  ]);
var html = Thermite_Html.createElement("html");
var html$prime = html([  ]);
var hr = Thermite_Html.createElement("hr");
var hr$prime = hr([  ]);
var header = Thermite_Html.createElement("header");
var header$prime = header([  ]);
var head = Thermite_Html.createElement("head");
var head$prime = head([  ]);
var h6 = Thermite_Html.createElement("h6");
var h6$prime = h6([  ]);
var h5 = Thermite_Html.createElement("h5");
var h5$prime = h5([  ]);
var h4 = Thermite_Html.createElement("h4");
var h4$prime = h4([  ]);
var h3 = Thermite_Html.createElement("h3");
var h3$prime = h3([  ]);
var h2 = Thermite_Html.createElement("h2");
var h2$prime = h2([  ]);
var h1 = Thermite_Html.createElement("h1");
var h1$prime = h1([  ]);
var form = Thermite_Html.createElement("form");
var form$prime = form([  ]);
var footer = Thermite_Html.createElement("footer");
var footer$prime = footer([  ]);
var figure = Thermite_Html.createElement("figure");
var figure$prime = figure([  ]);
var figcaption = Thermite_Html.createElement("figcaption");
var figcaption$prime = figcaption([  ]);
var fieldset = Thermite_Html.createElement("fieldset");
var fieldset$prime = fieldset([  ]);
var embed = Thermite_Html.createElement("embed");
var embed$prime = embed([  ]);
var em = Thermite_Html.createElement("em");
var em$prime = em([  ]);
var dt = Thermite_Html.createElement("dt");
var dt$prime = dt([  ]);
var dl = Thermite_Html.createElement("dl");
var dl$prime = dl([  ]);
var div = Thermite_Html.createElement("div");
var div$prime = div([  ]);
var dialog = Thermite_Html.createElement("dialog");
var dialog$prime = dialog([  ]);
var dfn = Thermite_Html.createElement("dfn");
var dfn$prime = dfn([  ]);
var details = Thermite_Html.createElement("details");
var details$prime = details([  ]);
var del = Thermite_Html.createElement("del");
var del$prime = del([  ]);
var dd = Thermite_Html.createElement("dd");
var dd$prime = dd([  ]);
var datalist = Thermite_Html.createElement("datalist");
var datalist$prime = datalist([  ]);
var colgroup = Thermite_Html.createElement("colgroup");
var colgroup$prime = colgroup([  ]);
var col = Thermite_Html.createElement("col");
var col$prime = col([  ]);
var code = Thermite_Html.createElement("code");
var code$prime = code([  ]);
var cite = Thermite_Html.createElement("cite");
var cite$prime = cite([  ]);
var caption = Thermite_Html.createElement("caption");
var caption$prime = caption([  ]);
var canvas = Thermite_Html.createElement("canvas");
var canvas$prime = canvas([  ]);
var button = Thermite_Html.createElement("button");
var button$prime = button([  ]);
var br = Thermite_Html.createElement("br");
var br$prime = br([  ]);
var body = Thermite_Html.createElement("body");
var body$prime = body([  ]);
var link$prime = body([  ]);
var blockquote = Thermite_Html.createElement("blockquote");
var blockquote$prime = blockquote([  ]);
var big = Thermite_Html.createElement("big");
var big$prime = big([  ]);
var bdo = Thermite_Html.createElement("bdo");
var bdo$prime = bdo([  ]);
var bdi = Thermite_Html.createElement("bdi");
var bdi$prime = bdi([  ]);
var base = Thermite_Html.createElement("base");
var base$prime = base([  ]);
var b = Thermite_Html.createElement("b");
var b$prime = b([  ]);
var audio = Thermite_Html.createElement("audio");
var audio$prime = audio([  ]);
var aside = Thermite_Html.createElement("aside");
var aside$prime = aside([  ]);
var article = Thermite_Html.createElement("article");
var article$prime = article([  ]);
var area = Thermite_Html.createElement("area");
var area$prime = area([  ]);
var address = Thermite_Html.createElement("address");
var address$prime = address([  ]);
var abbr = Thermite_Html.createElement("abbr");
var abbr$prime = abbr([  ]);
var a = Thermite_Html.createElement("a");
var a$prime = a([  ]);
var _data = Thermite_Html.createElement("data");
var _data$prime = _data([  ]);
module.exports = {
    _data: _data, 
    "_data'": _data$prime, 
    a: a, 
    "a'": a$prime, 
    abbr: abbr, 
    "abbr'": abbr$prime, 
    address: address, 
    "address'": address$prime, 
    area: area, 
    "area'": area$prime, 
    article: article, 
    "article'": article$prime, 
    aside: aside, 
    "aside'": aside$prime, 
    audio: audio, 
    "audio'": audio$prime, 
    b: b, 
    "b'": b$prime, 
    base: base, 
    "base'": base$prime, 
    bdi: bdi, 
    "bdi'": bdi$prime, 
    bdo: bdo, 
    "bdo'": bdo$prime, 
    big: big, 
    "big'": big$prime, 
    blockquote: blockquote, 
    "blockquote'": blockquote$prime, 
    body: body, 
    "body'": body$prime, 
    br: br, 
    "br'": br$prime, 
    button: button, 
    "button'": button$prime, 
    canvas: canvas, 
    "canvas'": canvas$prime, 
    caption: caption, 
    "caption'": caption$prime, 
    cite: cite, 
    "cite'": cite$prime, 
    code: code, 
    "code'": code$prime, 
    col: col, 
    "col'": col$prime, 
    colgroup: colgroup, 
    "colgroup'": colgroup$prime, 
    datalist: datalist, 
    "datalist'": datalist$prime, 
    dd: dd, 
    "dd'": dd$prime, 
    del: del, 
    "del'": del$prime, 
    details: details, 
    "details'": details$prime, 
    dfn: dfn, 
    "dfn'": dfn$prime, 
    dialog: dialog, 
    "dialog'": dialog$prime, 
    div: div, 
    "div'": div$prime, 
    dl: dl, 
    "dl'": dl$prime, 
    dt: dt, 
    "dt'": dt$prime, 
    em: em, 
    "em'": em$prime, 
    embed: embed, 
    "embed'": embed$prime, 
    fieldset: fieldset, 
    "fieldset'": fieldset$prime, 
    figcaption: figcaption, 
    "figcaption'": figcaption$prime, 
    figure: figure, 
    "figure'": figure$prime, 
    footer: footer, 
    "footer'": footer$prime, 
    form: form, 
    "form'": form$prime, 
    h1: h1, 
    "h1'": h1$prime, 
    h2: h2, 
    "h2'": h2$prime, 
    h3: h3, 
    "h3'": h3$prime, 
    h4: h4, 
    "h4'": h4$prime, 
    h5: h5, 
    "h5'": h5$prime, 
    h6: h6, 
    "h6'": h6$prime, 
    head: head, 
    "head'": head$prime, 
    header: header, 
    "header'": header$prime, 
    hr: hr, 
    "hr'": hr$prime, 
    html: html, 
    "html'": html$prime, 
    i: i, 
    "i'": i$prime, 
    iframe: iframe, 
    "iframe'": iframe$prime, 
    img: img, 
    "img'": img$prime, 
    input: input, 
    "input'": input$prime, 
    ins: ins, 
    "ins'": ins$prime, 
    kbd: kbd, 
    "kbd'": kbd$prime, 
    keygen: keygen, 
    "keygen'": keygen$prime, 
    label: label, 
    "label'": label$prime, 
    legend: legend, 
    "legend'": legend$prime, 
    li: li, 
    "li'": li$prime, 
    link: link, 
    "link'": link$prime, 
    main: main, 
    "main'": main$prime, 
    map: map, 
    "map'": map$prime, 
    mark: mark, 
    "mark'": mark$prime, 
    menu: menu, 
    "menu'": menu$prime, 
    menuitem: menuitem, 
    "menuitem'": menuitem$prime, 
    meta: meta, 
    "meta'": meta$prime, 
    meter: meter, 
    "meter'": meter$prime, 
    nav: nav, 
    "nav'": nav$prime, 
    noscript: noscript, 
    "noscript'": noscript$prime, 
    object: object, 
    "object'": object$prime, 
    ol: ol, 
    "ol'": ol$prime, 
    optgroup: optgroup, 
    "optgroup'": optgroup$prime, 
    option: option, 
    "option'": option$prime, 
    output: output, 
    "output'": output$prime, 
    p: p, 
    "p'": p$prime, 
    param: param, 
    "param'": param$prime, 
    picture: picture, 
    "picture'": picture$prime, 
    pre: pre, 
    "pre'": pre$prime, 
    progress: progress, 
    "progress'": progress$prime, 
    q: q, 
    "q'": q$prime, 
    rp: rp, 
    "rp'": rp$prime, 
    rt: rt, 
    "rt'": rt$prime, 
    ruby: ruby, 
    "ruby'": ruby$prime, 
    s: s, 
    "s'": s$prime, 
    samp: samp, 
    "samp'": samp$prime, 
    script: script, 
    "script'": script$prime, 
    section: section, 
    "section'": section$prime, 
    select: select, 
    "select'": select$prime, 
    small: small, 
    "small'": small$prime, 
    source: source, 
    "source'": source$prime, 
    span: span, 
    "span'": span$prime, 
    strong: strong, 
    "strong'": strong$prime, 
    style: style, 
    "style'": style$prime, 
    sub: sub, 
    "sub'": sub$prime, 
    summary: summary, 
    "summary'": summary$prime, 
    sup: sup, 
    "sup'": sup$prime, 
    table: table, 
    "table'": table$prime, 
    tbody: tbody, 
    "tbody'": tbody$prime, 
    td: td, 
    "td'": td$prime, 
    textarea: textarea, 
    "textarea'": textarea$prime, 
    tfoot: tfoot, 
    "tfoot'": tfoot$prime, 
    th: th, 
    "th'": th$prime, 
    thead: thead, 
    "thead'": thead$prime, 
    time: time, 
    "time'": time$prime, 
    title: title, 
    "title'": title$prime, 
    tr: tr, 
    "tr'": tr$prime, 
    track: track, 
    "track'": track$prime, 
    u: u, 
    "u'": u$prime, 
    ul: ul, 
    "ul'": ul$prime, 
    "var": $$var, 
    "var'": var$prime, 
    video: video, 
    "video'": video$prime, 
    wbr: wbr, 
    "wbr'": wbr$prime
};
