// Thermite.Html.Elements/index.js
"use strict";
var Thermite_Internal = require("Thermite.Internal");
var Prelude = require("Prelude");
var wbr = Thermite_Internal.createElementImpl("body");
var wbr$prime = wbr([  ]);
var video = Thermite_Internal.createElementImpl("video");
var video$prime = video([  ]);
var $$var = Thermite_Internal.createElementImpl("var");
var var$prime = $$var([  ]);
var ul = Thermite_Internal.createElementImpl("ul");
var ul$prime = ul([  ]);
var u = Thermite_Internal.createElementImpl("u");
var u$prime = u([  ]);
var track = Thermite_Internal.createElementImpl("track");
var track$prime = track([  ]);
var tr = Thermite_Internal.createElementImpl("tr");
var tr$prime = tr([  ]);
var title = Thermite_Internal.createElementImpl("title");
var title$prime = title([  ]);
var time = Thermite_Internal.createElementImpl("time");
var time$prime = time([  ]);
var thead = Thermite_Internal.createElementImpl("thead");
var thead$prime = thead([  ]);
var th = Thermite_Internal.createElementImpl("th");
var th$prime = th([  ]);
var tfoot = Thermite_Internal.createElementImpl("tfoot");
var tfoot$prime = tfoot([  ]);
var textarea = Thermite_Internal.createElementImpl("textarea");
var textarea$prime = textarea([  ]);
var td = Thermite_Internal.createElementImpl("td");
var td$prime = td([  ]);
var tbody = Thermite_Internal.createElementImpl("tbody");
var tbody$prime = tbody([  ]);
var table = Thermite_Internal.createElementImpl("table");
var table$prime = table([  ]);
var sup = Thermite_Internal.createElementImpl("sup");
var sup$prime = sup([  ]);
var summary = Thermite_Internal.createElementImpl("summary");
var summary$prime = summary([  ]);
var sub = Thermite_Internal.createElementImpl("sub");
var sub$prime = sub([  ]);
var style = Thermite_Internal.createElementImpl("style");
var style$prime = style([  ]);
var strong = Thermite_Internal.createElementImpl("strong");
var strong$prime = strong([  ]);
var span = Thermite_Internal.createElementImpl("span");
var span$prime = span([  ]);
var source = Thermite_Internal.createElementImpl("source");
var source$prime = source([  ]);
var small = Thermite_Internal.createElementImpl("small");
var small$prime = small([  ]);
var select = Thermite_Internal.createElementImpl("select");
var select$prime = select([  ]);
var section = Thermite_Internal.createElementImpl("section");
var section$prime = section([  ]);
var script = Thermite_Internal.createElementImpl("script");
var script$prime = script([  ]);
var samp = Thermite_Internal.createElementImpl("samp");
var samp$prime = samp([  ]);
var s = Thermite_Internal.createElementImpl("s");
var s$prime = s([  ]);
var ruby = Thermite_Internal.createElementImpl("ruby");
var ruby$prime = ruby([  ]);
var rt = Thermite_Internal.createElementImpl("rt");
var rt$prime = rt([  ]);
var rp = Thermite_Internal.createElementImpl("rp");
var rp$prime = rp([  ]);
var q = Thermite_Internal.createElementImpl("q");
var q$prime = q([  ]);
var progress = Thermite_Internal.createElementImpl("progress");
var progress$prime = progress([  ]);
var pre = Thermite_Internal.createElementImpl("pre");
var pre$prime = pre([  ]);
var picture = Thermite_Internal.createElementImpl("picture");
var picture$prime = picture([  ]);
var param = Thermite_Internal.createElementImpl("param");
var param$prime = param([  ]);
var p = Thermite_Internal.createElementImpl("p");
var p$prime = p([  ]);
var output = Thermite_Internal.createElementImpl("output");
var output$prime = output([  ]);
var option = Thermite_Internal.createElementImpl("option");
var option$prime = option([  ]);
var optgroup = Thermite_Internal.createElementImpl("optgroup");
var optgroup$prime = optgroup([  ]);
var ol = Thermite_Internal.createElementImpl("ol");
var ol$prime = ol([  ]);
var object = Thermite_Internal.createElementImpl("object");
var object$prime = object([  ]);
var noscript = Thermite_Internal.createElementImpl("noscript");
var noscript$prime = noscript([  ]);
var nav = Thermite_Internal.createElementImpl("nav");
var nav$prime = nav([  ]);
var meter = Thermite_Internal.createElementImpl("meter");
var meter$prime = meter([  ]);
var meta = Thermite_Internal.createElementImpl("meta");
var meta$prime = meta([  ]);
var menuitem = Thermite_Internal.createElementImpl("menuitem");
var menuitem$prime = menuitem([  ]);
var menu = Thermite_Internal.createElementImpl("menu");
var menu$prime = menu([  ]);
var mark = Thermite_Internal.createElementImpl("mark");
var mark$prime = mark([  ]);
var map = Thermite_Internal.createElementImpl("map");
var map$prime = map([  ]);
var main = Thermite_Internal.createElementImpl("main");
var main$prime = main([  ]);
var link = Thermite_Internal.createElementImpl("link");
var li = Thermite_Internal.createElementImpl("li");
var li$prime = li([  ]);
var legend = Thermite_Internal.createElementImpl("legend");
var legend$prime = legend([  ]);
var label = Thermite_Internal.createElementImpl("label");
var label$prime = label([  ]);
var keygen = Thermite_Internal.createElementImpl("keygen");
var keygen$prime = keygen([  ]);
var kbd = Thermite_Internal.createElementImpl("kbd");
var kbd$prime = kbd([  ]);
var ins = Thermite_Internal.createElementImpl("ins");
var ins$prime = ins([  ]);
var input = Thermite_Internal.createElementImpl("input");
var input$prime = input([  ]);
var img = Thermite_Internal.createElementImpl("img");
var img$prime = img([  ]);
var iframe = Thermite_Internal.createElementImpl("iframe");
var iframe$prime = iframe([  ]);
var i = Thermite_Internal.createElementImpl("i");
var i$prime = i([  ]);
var html = Thermite_Internal.createElementImpl("html");
var html$prime = html([  ]);
var hr = Thermite_Internal.createElementImpl("hr");
var hr$prime = hr([  ]);
var header = Thermite_Internal.createElementImpl("header");
var header$prime = header([  ]);
var head = Thermite_Internal.createElementImpl("head");
var head$prime = head([  ]);
var h6 = Thermite_Internal.createElementImpl("h6");
var h6$prime = h6([  ]);
var h5 = Thermite_Internal.createElementImpl("h5");
var h5$prime = h5([  ]);
var h4 = Thermite_Internal.createElementImpl("h4");
var h4$prime = h4([  ]);
var h3 = Thermite_Internal.createElementImpl("h3");
var h3$prime = h3([  ]);
var h2 = Thermite_Internal.createElementImpl("h2");
var h2$prime = h2([  ]);
var h1 = Thermite_Internal.createElementImpl("h1");
var h1$prime = h1([  ]);
var form = Thermite_Internal.createElementImpl("form");
var form$prime = form([  ]);
var footer = Thermite_Internal.createElementImpl("footer");
var footer$prime = footer([  ]);
var figure = Thermite_Internal.createElementImpl("figure");
var figure$prime = figure([  ]);
var figcaption = Thermite_Internal.createElementImpl("figcaption");
var figcaption$prime = figcaption([  ]);
var fieldset = Thermite_Internal.createElementImpl("fieldset");
var fieldset$prime = fieldset([  ]);
var embed = Thermite_Internal.createElementImpl("embed");
var embed$prime = embed([  ]);
var em = Thermite_Internal.createElementImpl("em");
var em$prime = em([  ]);
var dt = Thermite_Internal.createElementImpl("dt");
var dt$prime = dt([  ]);
var dl = Thermite_Internal.createElementImpl("dl");
var dl$prime = dl([  ]);
var div = Thermite_Internal.createElementImpl("div");
var div$prime = div([  ]);
var dialog = Thermite_Internal.createElementImpl("dialog");
var dialog$prime = dialog([  ]);
var dfn = Thermite_Internal.createElementImpl("dfn");
var dfn$prime = dfn([  ]);
var details = Thermite_Internal.createElementImpl("details");
var details$prime = details([  ]);
var del = Thermite_Internal.createElementImpl("del");
var del$prime = del([  ]);
var dd = Thermite_Internal.createElementImpl("dd");
var dd$prime = dd([  ]);
var datalist = Thermite_Internal.createElementImpl("datalist");
var datalist$prime = datalist([  ]);
var colgroup = Thermite_Internal.createElementImpl("colgroup");
var colgroup$prime = colgroup([  ]);
var col = Thermite_Internal.createElementImpl("col");
var col$prime = col([  ]);
var code = Thermite_Internal.createElementImpl("code");
var code$prime = code([  ]);
var cite = Thermite_Internal.createElementImpl("cite");
var cite$prime = cite([  ]);
var caption = Thermite_Internal.createElementImpl("caption");
var caption$prime = caption([  ]);
var canvas = Thermite_Internal.createElementImpl("canvas");
var canvas$prime = canvas([  ]);
var button = Thermite_Internal.createElementImpl("button");
var button$prime = button([  ]);
var br = Thermite_Internal.createElementImpl("br");
var br$prime = br([  ]);
var body = Thermite_Internal.createElementImpl("body");
var body$prime = body([  ]);
var link$prime = body([  ]);
var blockquote = Thermite_Internal.createElementImpl("blockquote");
var blockquote$prime = blockquote([  ]);
var big = Thermite_Internal.createElementImpl("big");
var big$prime = big([  ]);
var bdo = Thermite_Internal.createElementImpl("bdo");
var bdo$prime = bdo([  ]);
var bdi = Thermite_Internal.createElementImpl("bdi");
var bdi$prime = bdi([  ]);
var base = Thermite_Internal.createElementImpl("base");
var base$prime = base([  ]);
var b = Thermite_Internal.createElementImpl("b");
var b$prime = b([  ]);
var audio = Thermite_Internal.createElementImpl("audio");
var audio$prime = audio([  ]);
var aside = Thermite_Internal.createElementImpl("aside");
var aside$prime = aside([  ]);
var article = Thermite_Internal.createElementImpl("article");
var article$prime = article([  ]);
var area = Thermite_Internal.createElementImpl("area");
var area$prime = area([  ]);
var address = Thermite_Internal.createElementImpl("address");
var address$prime = address([  ]);
var abbr = Thermite_Internal.createElementImpl("abbr");
var abbr$prime = abbr([  ]);
var a = Thermite_Internal.createElementImpl("a");
var a$prime = a([  ]);
var _data = Thermite_Internal.createElementImpl("data");
var _data$prime = _data([  ]);
module.exports = {
    _data: _data, 
    "_data'": _data$prime, 
    a: a, 
    "a'": a$prime, 
    abbr: abbr, 
    "abbr'": abbr$prime, 
    address: address, 
    "address'": address$prime, 
    area: area, 
    "area'": area$prime, 
    article: article, 
    "article'": article$prime, 
    aside: aside, 
    "aside'": aside$prime, 
    audio: audio, 
    "audio'": audio$prime, 
    b: b, 
    "b'": b$prime, 
    base: base, 
    "base'": base$prime, 
    bdi: bdi, 
    "bdi'": bdi$prime, 
    bdo: bdo, 
    "bdo'": bdo$prime, 
    big: big, 
    "big'": big$prime, 
    blockquote: blockquote, 
    "blockquote'": blockquote$prime, 
    body: body, 
    "body'": body$prime, 
    br: br, 
    "br'": br$prime, 
    button: button, 
    "button'": button$prime, 
    canvas: canvas, 
    "canvas'": canvas$prime, 
    caption: caption, 
    "caption'": caption$prime, 
    cite: cite, 
    "cite'": cite$prime, 
    code: code, 
    "code'": code$prime, 
    col: col, 
    "col'": col$prime, 
    colgroup: colgroup, 
    "colgroup'": colgroup$prime, 
    datalist: datalist, 
    "datalist'": datalist$prime, 
    dd: dd, 
    "dd'": dd$prime, 
    del: del, 
    "del'": del$prime, 
    details: details, 
    "details'": details$prime, 
    dfn: dfn, 
    "dfn'": dfn$prime, 
    dialog: dialog, 
    "dialog'": dialog$prime, 
    div: div, 
    "div'": div$prime, 
    dl: dl, 
    "dl'": dl$prime, 
    dt: dt, 
    "dt'": dt$prime, 
    em: em, 
    "em'": em$prime, 
    embed: embed, 
    "embed'": embed$prime, 
    fieldset: fieldset, 
    "fieldset'": fieldset$prime, 
    figcaption: figcaption, 
    "figcaption'": figcaption$prime, 
    figure: figure, 
    "figure'": figure$prime, 
    footer: footer, 
    "footer'": footer$prime, 
    form: form, 
    "form'": form$prime, 
    h1: h1, 
    "h1'": h1$prime, 
    h2: h2, 
    "h2'": h2$prime, 
    h3: h3, 
    "h3'": h3$prime, 
    h4: h4, 
    "h4'": h4$prime, 
    h5: h5, 
    "h5'": h5$prime, 
    h6: h6, 
    "h6'": h6$prime, 
    head: head, 
    "head'": head$prime, 
    header: header, 
    "header'": header$prime, 
    hr: hr, 
    "hr'": hr$prime, 
    html: html, 
    "html'": html$prime, 
    i: i, 
    "i'": i$prime, 
    iframe: iframe, 
    "iframe'": iframe$prime, 
    img: img, 
    "img'": img$prime, 
    input: input, 
    "input'": input$prime, 
    ins: ins, 
    "ins'": ins$prime, 
    kbd: kbd, 
    "kbd'": kbd$prime, 
    keygen: keygen, 
    "keygen'": keygen$prime, 
    label: label, 
    "label'": label$prime, 
    legend: legend, 
    "legend'": legend$prime, 
    li: li, 
    "li'": li$prime, 
    link: link, 
    "link'": link$prime, 
    main: main, 
    "main'": main$prime, 
    map: map, 
    "map'": map$prime, 
    mark: mark, 
    "mark'": mark$prime, 
    menu: menu, 
    "menu'": menu$prime, 
    menuitem: menuitem, 
    "menuitem'": menuitem$prime, 
    meta: meta, 
    "meta'": meta$prime, 
    meter: meter, 
    "meter'": meter$prime, 
    nav: nav, 
    "nav'": nav$prime, 
    noscript: noscript, 
    "noscript'": noscript$prime, 
    object: object, 
    "object'": object$prime, 
    ol: ol, 
    "ol'": ol$prime, 
    optgroup: optgroup, 
    "optgroup'": optgroup$prime, 
    option: option, 
    "option'": option$prime, 
    output: output, 
    "output'": output$prime, 
    p: p, 
    "p'": p$prime, 
    param: param, 
    "param'": param$prime, 
    picture: picture, 
    "picture'": picture$prime, 
    pre: pre, 
    "pre'": pre$prime, 
    progress: progress, 
    "progress'": progress$prime, 
    q: q, 
    "q'": q$prime, 
    rp: rp, 
    "rp'": rp$prime, 
    rt: rt, 
    "rt'": rt$prime, 
    ruby: ruby, 
    "ruby'": ruby$prime, 
    s: s, 
    "s'": s$prime, 
    samp: samp, 
    "samp'": samp$prime, 
    script: script, 
    "script'": script$prime, 
    section: section, 
    "section'": section$prime, 
    select: select, 
    "select'": select$prime, 
    small: small, 
    "small'": small$prime, 
    source: source, 
    "source'": source$prime, 
    span: span, 
    "span'": span$prime, 
    strong: strong, 
    "strong'": strong$prime, 
    style: style, 
    "style'": style$prime, 
    sub: sub, 
    "sub'": sub$prime, 
    summary: summary, 
    "summary'": summary$prime, 
    sup: sup, 
    "sup'": sup$prime, 
    table: table, 
    "table'": table$prime, 
    tbody: tbody, 
    "tbody'": tbody$prime, 
    td: td, 
    "td'": td$prime, 
    textarea: textarea, 
    "textarea'": textarea$prime, 
    tfoot: tfoot, 
    "tfoot'": tfoot$prime, 
    th: th, 
    "th'": th$prime, 
    thead: thead, 
    "thead'": thead$prime, 
    time: time, 
    "time'": time$prime, 
    title: title, 
    "title'": title$prime, 
    tr: tr, 
    "tr'": tr$prime, 
    track: track, 
    "track'": track$prime, 
    u: u, 
    "u'": u$prime, 
    ul: ul, 
    "ul'": ul$prime, 
    "var": $$var, 
    "var'": var$prime, 
    video: video, 
    "video'": video$prime, 
    wbr: wbr, 
    "wbr'": wbr$prime
};
