// Control.Comonad.Store.Trans/index.js
"use strict";
var Control_Comonad = require("Control.Comonad");
var Prelude = require("Prelude");
var Data_Tuple = require("Data.Tuple");
var Control_Extend = require("Control.Extend");
var Control_Comonad_Trans = require("Control.Comonad.Trans");
var StoreT = {
    create: function (value) {
        return value;
    }
};
var runStoreT = function (_199) {
    return _199;
};
var functorStoreT = function (__dict_Functor_0) {
    return new Prelude.Functor(function (_200) {
        return function (_201) {
            return StoreT.create(new Data_Tuple.Tuple(Prelude["<$>"](__dict_Functor_0)(function (h) {
                return Prelude[">>>"](Prelude.semigroupoidArr)(h)(_200);
            })(_201.value0), _201.value1));
        };
    });
};
var extendStoreT = function (__dict_Extend_1) {
    return new Control_Extend.Extend(function (_202) {
        return function (_203) {
            return StoreT.create(new Data_Tuple.Tuple(Control_Extend["<<="](__dict_Extend_1)(function (w$prime) {
                return function (s$prime) {
                    return _202(StoreT.create(new Data_Tuple.Tuple(w$prime, s$prime)));
                };
            })(_203.value0), _203.value1));
        };
    }, function () {
        return functorStoreT(__dict_Extend_1["__superclass_Prelude.Functor_0"]());
    });
};
var comonadTransStoreT = new Control_Comonad_Trans.ComonadTrans(function (__dict_Comonad_2) {
    return function (_205) {
        return Prelude["<$>"]((__dict_Comonad_2["__superclass_Control.Extend.Extend_0"]())["__superclass_Prelude.Functor_0"]())(function (f) {
            return f(_205.value1);
        })(_205.value0);
    };
});
var comonadStoreT = function (__dict_Comonad_3) {
    return new Control_Comonad.Comonad(function () {
        return extendStoreT(__dict_Comonad_3["__superclass_Control.Extend.Extend_0"]());
    }, function (_204) {
        return Control_Comonad.extract(__dict_Comonad_3)(_204.value0)(_204.value1);
    });
};
module.exports = {
    StoreT: StoreT, 
    comonadStoreT: comonadStoreT, 
    comonadTransStoreT: comonadTransStoreT, 
    extendStoreT: extendStoreT, 
    functorStoreT: functorStoreT, 
    runStoreT: runStoreT
};
