// Data.Foreign.Index/index.js
"use strict";
var Data_Function = require("Data.Function");
var Data_Foreign = require("Data.Foreign");
var Prelude = require("Prelude");
var Data_Either = require("Data.Either");
function Index($bang, errorAt, hasOwnProperty, hasProperty) {
    this["!"] = $bang;
    this.errorAt = errorAt;
    this.hasOwnProperty = hasOwnProperty;
    this.hasProperty = hasProperty;
};

  function unsafeReadPropImpl(f, s, key, value) {
    return value == null ? f : s(value[key]);
  }
  ;

  function unsafeHasOwnProperty(prop, value) {
    return Object.prototype.hasOwnProperty.call(value, prop);
  }
  ;

  function unsafeHasProperty(prop, value) {
    return prop in value;
  }
  ;
var $bang = function (dict) {
    return dict["!"];
};
var unsafeReadProp = function (k) {
    return function (value) {
        return unsafeReadPropImpl(new Data_Either.Left(new Data_Foreign.TypeMismatch("object", Data_Foreign.typeOf(value))), Prelude.pure(Data_Either.applicativeEither), k, value);
    };
};
var prop = unsafeReadProp;
var index = unsafeReadProp;
var hasPropertyImpl = function (_101) {
    return function (_102) {
        if (Data_Foreign.isNull(_102)) {
            return false;
        };
        if (Data_Foreign.isUndefined(_102)) {
            return false;
        };
        if (Data_Foreign.typeOf(_102) === "object" || Data_Foreign.typeOf(_102) === "function") {
            return unsafeHasProperty(_101, _102);
        };
        return false;
    };
};
var hasProperty = function (dict) {
    return dict.hasProperty;
};
var hasOwnPropertyImpl = function (_99) {
    return function (_100) {
        if (Data_Foreign.isNull(_100)) {
            return false;
        };
        if (Data_Foreign.isUndefined(_100)) {
            return false;
        };
        if (Data_Foreign.typeOf(_100) === "object" || Data_Foreign.typeOf(_100) === "function") {
            return unsafeHasOwnProperty(_99, _100);
        };
        return false;
    };
};
var indexNumber = new Index(Prelude.flip(index), Data_Foreign.ErrorAtIndex.create, hasOwnPropertyImpl, hasPropertyImpl);
var indexString = new Index(Prelude.flip(prop), Data_Foreign.ErrorAtProperty.create, hasOwnPropertyImpl, hasPropertyImpl);
var hasOwnProperty = function (dict) {
    return dict.hasOwnProperty;
};
var errorAt = function (dict) {
    return dict.errorAt;
};
module.exports = {
    "!": $bang, 
    Index: Index, 
    errorAt: errorAt, 
    hasOwnProperty: hasOwnProperty, 
    hasProperty: hasProperty, 
    index: index, 
    indexNumber: indexNumber, 
    indexString: indexString, 
    prop: prop
};
