// Data.Identity/index.js
"use strict";
var Prelude = require("Prelude");
var Data_Foldable = require("Data.Foldable");
var Data_Traversable = require("Data.Traversable");
var Control_Extend = require("Control.Extend");
var Control_Comonad = require("Control.Comonad");
var Identity = {
    create: function (value) {
        return value;
    }
};
var showConst = function (__dict_Show_2) {
    return new Prelude.Show(function (_307) {
        return "Identity (" + (Prelude.show(__dict_Show_2)(_307) + ")");
    });
};
var runIdentity = function (_302) {
    return _302;
};
var functorIdentity = new Prelude.Functor(function (_308) {
    return function (_309) {
        return _308(_309);
    };
});
var foldableIdentity = new Data_Foldable.Foldable(function (__dict_Monoid_4) {
    return function (_319) {
        return function (_320) {
            return _319(_320);
        };
    };
}, function (_316) {
    return function (_317) {
        return function (_318) {
            return _316(_317)(_318);
        };
    };
}, function (_313) {
    return function (_314) {
        return function (_315) {
            return _313(_315)(_314);
        };
    };
});
var traversableIdentity = new Data_Traversable.Traversable(function () {
    return foldableIdentity;
}, function () {
    return functorIdentity;
}, function (__dict_Applicative_1) {
    return function (_323) {
        return Prelude["<$>"]((__dict_Applicative_1["__superclass_Prelude.Apply_0"]())["__superclass_Prelude.Functor_0"]())(Identity.create)(_323);
    };
}, function (__dict_Applicative_0) {
    return function (_321) {
        return function (_322) {
            return Prelude["<$>"]((__dict_Applicative_0["__superclass_Prelude.Apply_0"]())["__superclass_Prelude.Functor_0"]())(Identity.create)(_321(_322));
        };
    };
});
var extendIdentity = new Control_Extend.Extend(function (f) {
    return function (m) {
        return f(m);
    };
}, function () {
    return functorIdentity;
});
var eqIdentity = function (__dict_Eq_5) {
    return new Prelude.Eq(function (x) {
        return function (y) {
            return !Prelude["=="](eqIdentity(__dict_Eq_5))(x)(y);
        };
    }, function (_303) {
        return function (_304) {
            return Prelude["=="](__dict_Eq_5)(_303)(_304);
        };
    });
};
var ordIdentity = function (__dict_Ord_3) {
    return new Prelude.Ord(function () {
        return eqIdentity(__dict_Ord_3["__superclass_Prelude.Eq_0"]());
    }, function (_305) {
        return function (_306) {
            return Prelude.compare(__dict_Ord_3)(_305)(_306);
        };
    });
};
var comonadIdentity = new Control_Comonad.Comonad(function () {
    return extendIdentity;
}, function (_312) {
    return _312;
});
var applyIdentity = new Prelude.Apply(function (_310) {
    return function (_311) {
        return _310(_311);
    };
}, function () {
    return functorIdentity;
});
var bindIdentity = new Prelude.Bind(function (m) {
    return function (f) {
        return f(runIdentity(m));
    };
}, function () {
    return applyIdentity;
});
var applicativeIdentity = new Prelude.Applicative(function () {
    return applyIdentity;
}, Identity.create);
var monadIdentity = new Prelude.Monad(function () {
    return applicativeIdentity;
}, function () {
    return bindIdentity;
});
module.exports = {
    Identity: Identity, 
    applicativeIdentity: applicativeIdentity, 
    applyIdentity: applyIdentity, 
    bindIdentity: bindIdentity, 
    comonadIdentity: comonadIdentity, 
    eqIdentity: eqIdentity, 
    extendIdentity: extendIdentity, 
    foldableIdentity: foldableIdentity, 
    functorIdentity: functorIdentity, 
    monadIdentity: monadIdentity, 
    ordIdentity: ordIdentity, 
    runIdentity: runIdentity, 
    showConst: showConst, 
    traversableIdentity: traversableIdentity
};
