// Control.Comonad.Traced.Trans/index.js
"use strict";
var Control_Comonad = require("Control.Comonad");
var Data_Monoid = require("Data.Monoid");
var Prelude = require("Prelude");
var Control_Extend = require("Control.Extend");
var Control_Comonad_Trans = require("Control.Comonad.Trans");
var TracedT = {
    create: function (value) {
        return value;
    }
};
var runTracedT = function (_209) {
    return _209;
};
var functorTracedT = function (__dict_Functor_0) {
    return new Prelude.Functor(function (_210) {
        return function (_211) {
            return Prelude["<$>"](__dict_Functor_0)(function (g) {
                return function (t) {
                    return _210(g(t));
                };
            })(_211);
        };
    });
};
var extendTracedT = function (__dict_Extend_1) {
    return function (__dict_Semigroup_2) {
        return new Control_Extend.Extend(function (_212) {
            return function (_213) {
                return Control_Extend["<<="](__dict_Extend_1)(function (w_1) {
                    return function (t) {
                        return _212(Prelude["<$>"](__dict_Extend_1["__superclass_Prelude.Functor_0"]())(function (h) {
                            return function (t$prime) {
                                return h(Prelude["<>"](__dict_Semigroup_2)(t)(t$prime));
                            };
                        })(w_1));
                    };
                })(_213);
            };
        }, function () {
            return functorTracedT(__dict_Extend_1["__superclass_Prelude.Functor_0"]());
        });
    };
};
var comonadTransTracedT = function (__dict_Monoid_3) {
    return new Control_Comonad_Trans.ComonadTrans(function (__dict_Comonad_4) {
        return function (_215) {
            return Prelude["<$>"]((__dict_Comonad_4["__superclass_Control.Extend.Extend_0"]())["__superclass_Prelude.Functor_0"]())(function (f) {
                return f(Data_Monoid.mempty(__dict_Monoid_3));
            })(_215);
        };
    });
};
var comonadTracedT = function (__dict_Comonad_5) {
    return function (__dict_Monoid_6) {
        return new Control_Comonad.Comonad(function () {
            return extendTracedT(__dict_Comonad_5["__superclass_Control.Extend.Extend_0"]())(__dict_Monoid_6["__superclass_Prelude.Semigroup_0"]());
        }, function (_214) {
            return Control_Comonad.extract(__dict_Comonad_5)(_214)(Data_Monoid.mempty(__dict_Monoid_6));
        });
    };
};
module.exports = {
    TracedT: TracedT, 
    comonadTracedT: comonadTracedT, 
    comonadTransTracedT: comonadTransTracedT, 
    extendTracedT: extendTracedT, 
    functorTracedT: functorTracedT, 
    runTracedT: runTracedT
};
