// Thermite/index.js
"use strict";
var Thermite_Internal = require("Thermite.Internal");
var Thermite_Action = require("Thermite.Action");
var Data_Maybe = require("Data.Maybe");
var Prelude = require("Prelude");
var Thermite_Types = require("Thermite.Types");
var simpleSpec = function (initialState) {
    return function (performAction) {
        return function (render_1) {
            return {
                initialState: initialState, 
                performAction: performAction, 
                render: render_1, 
                componentWillMount: Data_Maybe.Nothing.value, 
                displayName: Data_Maybe.Nothing.value
            };
        };
    };
};
var render = Thermite_Internal.renderImpl;
var displayName = function (_339) {
    return function (_340) {
        var _1081 = {};
        for (var _1082 in _340) {
            if (_340.hasOwnProperty(_1082)) {
                _1081[_1082] = _340[_1082];
            };
        };
        _1081.displayName = new Data_Maybe.Just(_339);
        return _1081;
    };
};
var createClass = Thermite_Internal.createClassImpl(Thermite_Action.runAction)(Data_Maybe.maybe);
var componentWillMount = function (_337) {
    return function (_338) {
        var _1085 = {};
        for (var _1086 in _338) {
            if (_338.hasOwnProperty(_1086)) {
                _1085[_1086] = _338[_1086];
            };
        };
        _1085.componentWillMount = new Data_Maybe.Just(_337);
        return _1085;
    };
};
module.exports = {
    componentWillMount: componentWillMount, 
    createClass: createClass, 
    displayName: displayName, 
    render: render, 
    simpleSpec: simpleSpec
};
