// Thermite.Events/index.js
"use strict";
var Prelude = require("Prelude");

  function event(name) {
    return function onClick(context) {
      return function(f) {
        return [name, function(e) {
          context.performAction(f(e));
        }];
      };
    };
  }
  ;
var onWheel = event("onTouchCancel");
var onTouchStart = event("onTouchStart");
var onTouchMove = event("onTouchMove");
var onTouchEnd = event("onTouchEnd");
var onTouchCancel = event("onTouchCancel");
var onSubmit = event("onSubmit");
var onScroll = event("onScroll");
var onPaste = event("onPaste");
var onMouseUp = event("onMouseUp");
var onMouseOver = event("onMouseOver");
var onMouseOut = event("onMouseOut");
var onMouseMove = event("onMouseMove");
var onMouseLeave = event("onMouseLeave");
var onMouseEnter = event("onMouseEnter");
var onMouseDown = event("onMouseDown");
var onKeyUp = event("onKeyUp");
var onKeyPress = event("onKeyPress");
var onKeyDown = event("onKeyDown");
var onInput = event("onInput");
var onFocus = event("onFocus");
var onDrop = event("onDrop");
var onDragStart = event("onDragStart");
var onDragOver = event("onDragOver");
var onDragLeave = event("onDragLeave");
var onDragExit = event("onDragExit");
var onDragEnter = event("onDragEnter");
var onDragEnd = event("onDragEnd");
var onDrag = event("onDrag");
var onDoubleClick = event("onDoubleClick");
var onCut = event("onCut");
var onCopy = event("onCopy");
var onClick = event("onClick");
var onChange = event("onChange");
var onBlur = event("onBlur");
module.exports = {
    event: event, 
    onBlur: onBlur, 
    onChange: onChange, 
    onClick: onClick, 
    onCopy: onCopy, 
    onCut: onCut, 
    onDoubleClick: onDoubleClick, 
    onDrag: onDrag, 
    onDragEnd: onDragEnd, 
    onDragEnter: onDragEnter, 
    onDragExit: onDragExit, 
    onDragLeave: onDragLeave, 
    onDragOver: onDragOver, 
    onDragStart: onDragStart, 
    onDrop: onDrop, 
    onFocus: onFocus, 
    onInput: onInput, 
    onKeyDown: onKeyDown, 
    onKeyPress: onKeyPress, 
    onKeyUp: onKeyUp, 
    onMouseDown: onMouseDown, 
    onMouseEnter: onMouseEnter, 
    onMouseLeave: onMouseLeave, 
    onMouseMove: onMouseMove, 
    onMouseOut: onMouseOut, 
    onMouseOver: onMouseOver, 
    onMouseUp: onMouseUp, 
    onPaste: onPaste, 
    onScroll: onScroll, 
    onSubmit: onSubmit, 
    onTouchCancel: onTouchCancel, 
    onTouchEnd: onTouchEnd, 
    onTouchMove: onTouchMove, 
    onTouchStart: onTouchStart, 
    onWheel: onWheel
};
