// Control.Monad.Writer.Class/index.js
"use strict";
var Prelude = require("Prelude");
var Control_Monad_Writer_Trans = require("Control.Monad.Writer.Trans");
var Control_Monad_Trans = require("Control.Monad.Trans");
var Control_Monad_Error_Trans = require("Control.Monad.Error.Trans");
var Control_Monad_Maybe_Trans = require("Control.Monad.Maybe.Trans");
var Control_Monad_State_Trans = require("Control.Monad.State.Trans");
var Control_Monad_Reader_Trans = require("Control.Monad.Reader.Trans");
var Control_Monad_RWS = require("Control.Monad.RWS");
var Data_Tuple = require("Data.Tuple");
function MonadWriter(listen, pass, writer) {
    this.listen = listen;
    this.pass = pass;
    this.writer = writer;
};
var writer = function (dict) {
    return dict.writer;
};
var tell = function (__dict_Monoid_0) {
    return function (__dict_Monad_1) {
        return function (__dict_MonadWriter_2) {
            return function (w) {
                return writer(__dict_MonadWriter_2)(new Data_Tuple.Tuple(Prelude.unit, w));
            };
        };
    };
};
var pass = function (dict) {
    return dict.pass;
};
var monadWriterWriterT = function (__dict_Monoid_3) {
    return function (__dict_Monad_4) {
        return new MonadWriter(function (m) {
            return Control_Monad_Writer_Trans.WriterT.create(Prelude[">>="](__dict_Monad_4["__superclass_Prelude.Bind_1"]())(Control_Monad_Writer_Trans.runWriterT(m))(function (_20) {
                return Prelude["return"](__dict_Monad_4)(new Data_Tuple.Tuple(new Data_Tuple.Tuple(_20.value0, _20.value1), _20.value1));
            }));
        }, function (m) {
            return Control_Monad_Writer_Trans.WriterT.create(Prelude[">>="](__dict_Monad_4["__superclass_Prelude.Bind_1"]())(Control_Monad_Writer_Trans.runWriterT(m))(function (_21) {
                return Prelude["return"](__dict_Monad_4)(new Data_Tuple.Tuple(_21.value0.value0, _21.value0.value1(_21.value1)));
            }));
        }, Prelude["<<<"](Prelude.semigroupoidArr)(Control_Monad_Writer_Trans.WriterT.create)(Prelude["return"](__dict_Monad_4)));
    };
};
var monadWriterRWST = function (__dict_Monad_5) {
    return function (__dict_Monoid_6) {
        return new MonadWriter(Control_Monad_RWS.listen(__dict_Monad_5), Control_Monad_RWS.pass(__dict_Monad_5), Control_Monad_RWS.writer(__dict_Monad_5["__superclass_Prelude.Applicative_0"]()));
    };
};
var listen = function (dict) {
    return dict.listen;
};
var listens = function (__dict_Monoid_7) {
    return function (__dict_Monad_8) {
        return function (__dict_MonadWriter_9) {
            return function (f) {
                return function (m) {
                    return Prelude[">>="](__dict_Monad_8["__superclass_Prelude.Bind_1"]())(listen(__dict_MonadWriter_9)(m))(function (_18) {
                        return Prelude["return"](__dict_Monad_8)(new Data_Tuple.Tuple(_18.value0, f(_18.value1)));
                    });
                };
            };
        };
    };
};
var monadWriterErrorT = function (__dict_Monad_10) {
    return function (__dict_Error_11) {
        return function (__dict_MonadWriter_12) {
            return new MonadWriter(Control_Monad_Error_Trans.liftListenError(__dict_Monad_10)(listen(__dict_MonadWriter_12)), Control_Monad_Error_Trans.liftPassError(__dict_Monad_10)(pass(__dict_MonadWriter_12)), function (wd) {
                return Control_Monad_Trans.lift(Control_Monad_Error_Trans.monadTransErrorT(__dict_Error_11))(__dict_Monad_10)(writer(__dict_MonadWriter_12)(wd));
            });
        };
    };
};
var monadWriterMaybeT = function (__dict_Monad_13) {
    return function (__dict_MonadWriter_14) {
        return new MonadWriter(Control_Monad_Maybe_Trans.liftListenMaybe(__dict_Monad_13)(listen(__dict_MonadWriter_14)), Control_Monad_Maybe_Trans.liftPassMaybe(__dict_Monad_13)(pass(__dict_MonadWriter_14)), function (wd) {
            return Control_Monad_Trans.lift(Control_Monad_Maybe_Trans.monadTransMaybeT)(__dict_Monad_13)(writer(__dict_MonadWriter_14)(wd));
        });
    };
};
var monadWriterReaderT = function (__dict_Monad_15) {
    return function (__dict_MonadWriter_16) {
        return new MonadWriter(Control_Monad_Reader_Trans.mapReaderT(listen(__dict_MonadWriter_16)), Control_Monad_Reader_Trans.mapReaderT(pass(__dict_MonadWriter_16)), function (wd) {
            return Control_Monad_Trans.lift(Control_Monad_Reader_Trans.monadTransReaderT)(__dict_Monad_15)(writer(__dict_MonadWriter_16)(wd));
        });
    };
};
var monadWriterStateT = function (__dict_Monad_17) {
    return function (__dict_MonadWriter_18) {
        return new MonadWriter(Control_Monad_State_Trans.liftListenState(__dict_Monad_17)(listen(__dict_MonadWriter_18)), Control_Monad_State_Trans.liftPassState(__dict_Monad_17)(pass(__dict_MonadWriter_18)), function (wd) {
            return Control_Monad_Trans.lift(Control_Monad_State_Trans.monadTransStateT)(__dict_Monad_17)(writer(__dict_MonadWriter_18)(wd));
        });
    };
};
var censor = function (__dict_Monoid_19) {
    return function (__dict_Monad_20) {
        return function (__dict_MonadWriter_21) {
            return function (f) {
                return function (m) {
                    return pass(__dict_MonadWriter_21)(Prelude[">>="](__dict_Monad_20["__superclass_Prelude.Bind_1"]())(m)(function (_19) {
                        return Prelude["return"](__dict_Monad_20)(new Data_Tuple.Tuple(_19, f));
                    }));
                };
            };
        };
    };
};
module.exports = {
    MonadWriter: MonadWriter, 
    censor: censor, 
    listen: listen, 
    listens: listens, 
    monadWriterErrorT: monadWriterErrorT, 
    monadWriterMaybeT: monadWriterMaybeT, 
    monadWriterRWST: monadWriterRWST, 
    monadWriterReaderT: monadWriterReaderT, 
    monadWriterStateT: monadWriterStateT, 
    monadWriterWriterT: monadWriterWriterT, 
    pass: pass, 
    tell: tell, 
    writer: writer
};
